### PR TITLE
Somewhat Refactor the Build documentation

### PR DESCRIPTION
**Pull request by yashpalgoyal1304**
Do NOT sqaush the commits please

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,8 +131,25 @@ $ docker run -it --volume "/usr/share/fonts:/fonts" --volume "$(pwd):/data" --us
 
 #### Nix
 
-The `nix` package manager is available as a standalone package manager on many platforms and is a viable way to run SILE.
-See usage in the [NixOS section](#nixos).
+The `nix` package manager is available as a standalone package manager on many platforms other than NixOS; like most Linux and BSD distributions, macOS, and even for Windows via WSL; and thus is a viable alternative way to run SILE.
```
Comment (alerque) [inline README.md:134]:
```suggestion
The `nix` package manager is available as a standalone package manager on many platforms other than NixOS including most Linux and BSD distributions, macOS, and even for Windows via WSL; and thus presents a viable alternative way to run SILE on most systems.
```

File: README.md [hunk 2]
```diff
@@ -149,8 +131,25 @@ $ docker run -it --volume "/usr/share/fonts:/fonts" --volume "$(pwd):/data" --us
...
+
+[Nix packages][nix] are available in both the stable and unstable channels.
+Because all packages are first made available in the unstable channel and then eventually make their way into the stable channel, to get the latest stable SILE releases we recommend hitting up the unstable channel first.
+You can use all the usual Nix tricks including launching a new shell with the `sile` command available or running it directly from any shell:
+
+```console
+$ nix shell nixos-unstable#sile
```
Comment (alerque) [inline README.md:141]: Unfortunately there are a lot of mixups between the legacy `nix-shell` and the new easier to use `nix shell` that isn't quite stable yet. The current invocation should be:

```suggestion
$ nix shell nixpkgs/nixpkgs-unstable#sile
```

File: README.md [hunk 3]
```diff
@@ -149,8 +131,25 @@ $ docker run -it --volume "/usr/share/fonts:/fonts" --volume "$(pwd):/data" --us
...
+$ sile <arguments>
+
+$ nix run nixos-unstable#sile -- <arguments>
```
Comment (alerque) [inline README.md:144]:
```suggestion
$ nix run nixpkgs/nixpkgs-unstable#sile -- <arguments>
```

File: README.md [hunk 4]
```diff
@@ -149,8 +131,25 @@ $ docker run -it --volume "/usr/share/fonts:/fonts" --volume "$(pwd):/data" --us
...
+```
+
+The SILE source repository is also a valid Nix [Flake][nix-flakes] which means you can run any specific version or the latest unreleased development code directly:
+
+```console
+$ nix run github:sile-typesetter/sile/v0.30.0 -- <sile arguments>
```
Comment (alerque) [inline README.md:150]:
```suggestion
$ nix run github:sile-typesetter/sile/v0.14.3 -- <sile arguments>
```

**Review by alerque — CHANGES_REQUESTED** [review]
Some of these suggestions are correcting issues with the refactoring, some (with the specific nix commands) are to update the names of the channels which changed upstream since this was written.

If you want I can make these changes and fixup the existing commit messages too (which need to be in Conventional Commit style).

File: README.md [hunk 5]
```diff
@@ -264,7 +263,7 @@ Note the comments in [the section about Docker](#docker) regarding version tags.
 ## Finding Out More
 
 Please read the [full SILE manual][doc] for more information about what SILE is and how it can help you.
-There are example documents (source and PDF) in the examples/ directory.
+There are example documents (source and PDF) in the [examples]/ directory.
```
[the remark this answers is not in the document]
Comment (yashpalgoyal1304) [reply README.md:266]: the relative hyperlink for the `[examples]` is already defined in the line `#L289`, so, i don't think repetition would be good here. how about the following?

```suggestion
There are example documents (source and PDF) in the [on the SILE website][examples].
```

**Comment by yashpalgoyal1304** [comment]
> If you want I can make these changes and fixup the existing commit messages too 

> (which need to be in Conventional Commit style).

hi, i am not sure i understood these lines (especially the last one). i mean yeah, thanks for using the suggestion feature - i have seen all the suggestions - and other than the ones i have already comment on, rest seem fine to me. so, i have no prob with you making changes too 😄 

oh, and also, i have one more thing to do regarding that CI thing. i'd do it now. so, don't merge it just yet.

**Comment by yashpalgoyal1304** [comment]
should i move the CI section at one higher level of heading?? (i.e. from level 3 to level 2)

i am unable to make suggestion regarding that, but it looks like this:

```diff
-### Use as a CI job
+## Use as a CI job

**Comment by alerque** [comment]
This will probably go faster if I just adjust from here, stand by...

**Comment by yashpalgoyal1304** [comment]
> @yashpalgoyal1304
[Merge branch 'build-documentation' of](https://github.com/sile-typesetter/sile/pull/1561/commits/99e73d7cff7ef45bacaf231a562588e4d1a367b1) [https://github.com/yashpalgoyal…](https://github.com/yashpalgoyal1304/sile)

whoops, did i do something wrong? 😅

**Comment by yashpalgoyal1304** [comment]
wow, magic 🪄 - i really don't know how git works lol 😅

**Review by alerque — APPROVED** [review]

**Pull request merged**